src/components/CategoryPage/CategoryPage.js

```diff
@@ -1,70 +1,32 @@
-import { useParams, Link } from "react-router-dom"
-import placeholder from '../../images/placeholder.png'
-import './CategoryPage.css'
+import { useParams, Link } from "react-router-dom";
+import placeholder from '../../images/placeholder.png';
+import './CategoryPage.css';
+import { getClosetData } from '../../apiCalls';
+import { useEffect, useState } from 'react';
 
 const CategoryPage = ({closeMenu}) => {
-  //this fake data will be removed after connecting the backend 
-  const allPieces = 
-    {
-      pants: [
-        {
-          id: 1,
-          image: placeholder
-        },
-        {
-          id: 2,
-          image: placeholder
-        }
-      ],
-      tops: [],
-      skirts: [
-        {
-          id: 3,
-          image: placeholder
-        }
-      ],
-      onePieces: [
-        {
-          id: 4,
-          image: placeholder
-        }
-      ],
-      shoes: [
-        {
-          id: 5,
-          image: placeholder
-        }
-      ],
-      bags: [
-        {
-          id: 6,
-          image: placeholder
-        }
-      ],
-      accessories: [
-        {
-          id: 7,
-          image: placeholder
-        }
-      ],
-      outerwear: [
-        {
-          id: 8,
-          image: placeholder
-        }
-      ],
-      miscellaneous: [
-        {
-          id: 9,
-          image: placeholder
-        }
-      ],
-    } 
   
-  //this will be a network request --> 
+  const [allPieces, setAllPieces] = useState(null);
   const category = useParams().category
-  const pieces = allPieces[category]
-  const pieceEls = pieces.map(piece => 
+ 
+    useEffect(() => {
+      const apiCall = async () => {
+        try {
+          let data = await getClosetData(category)
+          setAllPieces(data.filteredPieces)
+          return data
+        } catch (error) {
+          //should we move error up to app so that we can pass the same error state every where? or make it its own component?
```

Comment:
I think we should have an error state here so that this is isolated to represent this error at the location of the component rather than everywhere. We should probably also return a cleanup function that removes the error when the component unmounts